### PR TITLE
Fix dsym warnings during build macOS

### PR DESCRIPTION
**Pull request by ilyashumihin**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Fixed dsym warnings during build macOS application in release mode.

## Related PRs or issues

[AB#69480](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/69480)
[MacOS warnings PR](https://github.com/microsoft/appcenter-sdk-apple/issues/1789/)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 ### App Center
 
 * **[Fix]** Improve log messages for errors when it failed to read/write auth token history.
+* **[Fix]** Fixed some dsym warnings which could appear on macOS in release mode.
```
Comment (MatkovIvan) [inline CHANGELOG.md]:
```suggestion
* **[Fix]** Fixed some dSYM warnings which could appear on macOS in release mode.
```

File: Config/macOS.xcconfig [hunk 2]
```diff
@@ -5,3 +5,4 @@ VALID_ARCHS = $(ARCHS) i386
 SKIP_INSTALL = YES
 
 APPCENTER_BUILD_PLATFORM = macOS
+CLANG_ENABLE_MODULE_DEBUGGING = NO
```
Comment (MatkovIvan) [inline Config/macOS.xcconfig]: Please add the comment with explanation why we need it

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,7 @@
 ### App Center
 
 * **[Fix]** Improve log messages for errors when it failed to read/write auth token history.
+* **[Fix]** Fixed some dSYM warnings which could appear on macOS in release mode.
```
Comment (jaelim-ms) [inline CHANGELOG.md]:
```suggestion
* **[Fix]** Disable module debugging for release mode in the SDK to fix dSYM warnings.
```

File: Config/macOS.xcconfig (hunk 2 above)
Comment (jaelim-ms) [reply Config/macOS.xcconfig]: Shouldn't it be set only for Release?

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,7 @@
 ### App Center
 
 * **[Fix]** Improve log messages for errors when it failed to read/write auth token history.
+* **[Fix]** Disable module debugging for release mode in the SDK to fix dSYM warnings.
```
Comment (MatkovIvan) [inline CHANGELOG.md]: Merge develop changes and move it to next version

File: Config/macOS.xcconfig (hunk 2 above)
Comment (ilyashumihin) [reply Config/macOS.xcconfig]: Moved for release only

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,13 @@
 # App Center SDK for iOS, macOS and tvOS Change Log
 
+## Version 2.5.3
```
Comment (MatkovIvan) [inline CHANGELOG.md]:
```suggestion
## Version 2.5.3  (Under development)
```

**Pull request merged**